### ISSUE TITLE
Problems downloading an empty file

### ISSUE THREAD

**Issue by ottmar-zittlau** [eclipse-ecal/fineftp-server#64]
Hi,

I'm currently trying to update from v1.0.5 to v1.4.1 and experience some problems when downloading an empty file from the ftp server.

I've tried to debug this issue and found the following:
1. The error occurs here: https://github.com/eclipse-ecal/fineftp-server/blob/master/fineftp-server/src/ftp_session.cpp#L535
2. When I follow the `const auto file = ReadableFile::get(local_path);` call right before, I found that the line
https://github.com/eclipse-ecal/fineftp-server/blob/v1.4.1/fineftp-server/src/unix/file_man.cpp#L66 fails. 
3. I then checked the man-page of mmap (https://man7.org/linux/man-pages/man2/mmap.2.html) and found that
`The length argument specifies the length of the mapping (which must be greater than 0).` The variable `st.st_size` is actually 0, so I think, that this is the reason the call fails.

Would it make sense to exclude this check for empty files? 
Or do I have to configure something in my request (I'm using curl) to avoid this behaviour?
In v1.0.5 everything works fine.

I hacked something on my machine that fixes the issue for me:
```
  void* map_start = nullptr;
  if(st.st_size > 0){
  map_start = ::mmap(nullptr, st.st_size, PROT_READ, MAP_SHARED, handle, 0);
    if (MAP_FAILED == map_start)
    {
      ::close(handle);
      return {};
    }
  }
```
...but I'm a bit concerned that the null-pointer is put into the file struct later on.

Thanks and best regards
oz

**Comment by FlorianReimold** [comment]
Thanks for reporting! I was able to reproduce the issue on Windows, too. I created a GTest for that specific case which currently fails. But I guess it will be an easy fix with the code your provided.

**Comment by ottmar-zittlau** [comment]
Thanks for fixing this so quickly!

**Comment by FlorianReimold** [comment]
You are welcome!
The issue was introduced with fineFTP 1.4.0, where we switched to memory-mapped files for performance reasons. The old 1.0.5 used a different technique to read files, so that bug didn't exist there.

I released fineFTP 1.4.2, which solves the empty-file issue.
